src/postamble.js

```diff
@@ -195,7 +195,7 @@ function callMain(args) {
 #if ASYNCIFY
     // if we are saving the stack, then do not call exit, we are not
     // really exiting now, just unwinding the JS stack
-    if (!noExitRuntime) {
+    if (!keepRuntimeAlive()) {
```

Comment:
can callMain call `callUserCallback` to handle the exit etc? or would that increase code size or something?

Reply:
Perhaps has a followup?   It could decrease code size I imagine.